docs/extensions/generate_bug_description.py

```diff
@@ -49,5 +53,39 @@ def generate_bug_description(language):
                     bug_description_page.write('\n')
                 bug_description_page.write('\n')
 
+    with codecs.open('generated/detectorListEnabled.inc', 'w', encoding='UTF-8') as enabled_detector_page:
+        with codecs.open('generated/detectorListDisabled.inc', 'w', encoding='UTF-8') as disabled_detector_page:
+            for element in findbugs.iterfind(".//Detector"):
+                hidden = parse_bool_attr(element, "hidden")
+                if hidden:
+                    continue
+                klass = element.get("class")
+                disabled = parse_bool_attr(element, "disabled")
+                speed = element.get("speed")
+                reports = element.get("reports", "").split(",")
+
+                page = disabled_detector_page if disabled else enabled_detector_page
+
+                message = messages.find(".//Detector[@class='%s']" % klass)
+                short_name = klass.split(".")[-1]
+
+                page.write("%s\n%s\n\n" % (short_name, '^' * len(short_name)))
+
+                details = message.findtext('.//Details')
```

Comment:
This code throws an error like below if no translation is provided in `messages_ja.xml`:

```
Extension error (generate_bug_description):
Handler <function setup.<locals>.<lambda> at 0x7fc68a3d1550> for event 'builder-inited' threw an exception (exception: 'NoneType' object has no attribute 'findtext')
```

You can reproduce this error by following commands:

```sh
cd docs
docker build -t spotbugs-sphinx .
rm -rf .build
docker run -it -v $(pwd)/..:/docs spotbugs-sphinx make -e SPHINXOPTS="-D language='ja'" html
```

If you're using Windows, replace `$(pwd)/..` with fullpath of your SpotBugs root dir.

If possible, use the message in English (`messages.xml`) in case of no translation provided.
If it's difficult, it's OK to tell 'ja translation is not provided for foobar detector'.

Reply:
More or less fixed, I am using the English as a fallback but still, the process does raise a lot of warnings as sphinx complains about headings, several Japanese characters use 2 positions to print and the underlines are too short.
That problem is totally out of my capacity, I am a bit rusty in Python, and Unicode problems are a serious thing.